File: crates/tools/mudu_gen/src/src_check/fixtures/bad_call.c
/* Negative fixture: arity mismatch and unknown table. */
#include "mudu_sys.h"

static int get_balance(mudu_oid session, mudu_datum *args, mudu_rows *rows, mudu_error *err) {
    return mudu_query(session, "SELECT balance FROM wallets WHERE user_id = ?", args, 2,
                      rows, err);
}

static int delete_wallet(mudu_oid session, mudu_datum *args, mudu_rows *rows, mudu_error *err) {
    return mudu_command(session, "DELETE FROM wallet WHERE user_id = ?", args, 1, rows, err);
}

File: crates/tools/mudu_gen/src/src_check/fixtures/bad_call.cs
// Negative fixture: arity mismatch and unknown table.
internal static class Procedures
{
    public static long GetBalance(MuduOid session, long userId)
    {
        var rows = MuduSys.Query(session, "SELECT balance FROM wallets WHERE user_id = ?", userId, 0L);
        return rows;
    }

    public static long DeleteWallet(MuduOid session, long userId)
    {
        return MuduSys.Command(session, "DELETE FROM wallet WHERE user_id = ?", userId);
    }
}
